### DISCUSSION TITLE
How to connect to a mongodb replicaset topology with MongoDB engine?

### DISCUSSION THREAD

**Discussion by niltonvasques** [ClickHouse/ClickHouse#65101] [Q&A]
I tried for example to pass a list of host,port but it didn't worked.

`ENGINE = MongoDB('mongo1:25001,mongo2:25002', 'database', 'users', 'user', 'pwd')`

`DB::Exception: Received from localhost:9000. DB::Exception: Illegal port passed to function parseAddress: 30001,mongo2:30002. (BAD_ARGUMENTS)`

**Comment by UnamedRus** [comment]
Which version?

https://github.com/ClickHouse/ClickHouse/pull/46524

**Reply by niltonvasques** [reply]
Clickhouse Version `24.4.2` and mongo version `6.0` @UnamedRus

**Reply by UnamedRus** [reply]
>     if (theURI.getScheme() != "mongodb") throw Poco::UnknownURISchemeException(uri);

Tried? 
 mongodb://mongo1:25001,mongo2:25002/?replicaSet=myRepl&readPreference=primary

**Reply by niltonvasques** [reply]
Yes. Just tried again now:

```
CREATE TABLE my_table (_id String)
ENGINE = MongoDB('mongodb://mongo1:30001,mongo2:30002,mongo3:30003/?replicaSet=myReplica', 'prod-db', 'mycollection', 'myuser', 'mypass')


Received exception from server (version 24.4.2):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Illegal port passed to function parseAddress: //mongo1:30001,mongo2:30002,mongo3:30003/?replicaSet=myReplica&readPreference=secondary&connectTimeoutMS=10000. (BAD_ARGUMENTS)
```

Seems that it is not interpreting the first argument from MongoDB engine as a URI, it is only expecting a host:port format.

I saw the ticket and discussion from https://github.com/ClickHouse/ClickHouse/pull/46524 but I could not understand how to pass the URI to the engine from it. :disappointed:

**Reply by UnamedRus** [reply]
It's possible that it was implemented only for dictionaries, IE small/similar code change needs to be done to table function/engine

**Reply by niltonvasques** [reply]
that was also my feeling!
